File: Examples/VisualBasic/BarCode-Recognition/DetectOrientationOfBarCode.vb
Imports System.IO
Imports Aspose.BarCode
Imports System.Drawing.Printing
Imports Aspose.BarCodeRecognition

Namespace Aspose.BarCode.Examples.VisualBasic.BarCode.Recognition

    Public Class DetectOrientationOfBarCode
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_BarCodeRecognition()
            Dim dst As String = dataDir & Convert.ToString("Code39.png")

            ' Instantiate BarCodeReader object
            Dim reader As New BarCodeReader(dst, BarCodeReadType.Code39Standard)

            Try
                ' read Code39 bar code
                While reader.Read()
                    ' detect bar code orientation
                    System.Console.Write(Environment.NewLine + "Rotaion Angle: " + reader.GetAngle().ToString())
                End While
                reader.Close()

            Catch exp As Exception

                System.Console.Write(exp.Message)
            End Try

            Console.WriteLine(Environment.NewLine + "Barcode recognized from " & dst)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/BarCode-Recognition/DetectOrientationOfBarCode.cs
using System.IO;
using Aspose.BarCode;
using System;
using Aspose.BarCodeRecognition;

namespace Aspose.BarCode.Examples.CSharp.BarCode.Recognition
{
    class DetectOrientationOfBarCode
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_BarCodeRecognition();
            string dst = dataDir + "Code39.png";

            // Instantiate BarCodeReader object
            BarCodeReader reader = new BarCodeReader(dst, BarCodeReadType.Code39Standard);

            try
            {
                // read Code39 bar code
                while (reader.Read())
                {
                    // detect bar code orientation
                    System.Console.Write(Environment.NewLine + "Rotaion Angle: " + reader.GetAngle());
                }
                reader.Close();
            }

            catch (Exception exp)
            {

                System.Console.Write(exp.Message);
            }

            Console.WriteLine(Environment.NewLine + "Barcode recognized from " + dst);
        }
    }
}
